client/components/forms/form-password-input/index.jsx

```diff
@@ -33,27 +34,33 @@ class FormPasswordInput extends Component {
 	};
 
 	render() {
-		const toggleVisibilityClasses = clsx( {
-			'form-password-input__toggle': true,
-			'form-password-input__toggle-visibility': ! this.props.hideToggle,
-		} );
+		const { hideToggle, submitting, isHidden, translate, ...rest } = this.props;
+		const isValueHidden = this.getIsValueHidden();
 
-		/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */
 		return (
 			<div className="form-password-input">
 				<FormTextInput
 					autoComplete="off"
-					{ ...omit( this.props, 'hideToggle', 'submitting' ) }
+					{ ...rest }
 					ref={ this.textFieldRef }
-					type={ this.hidden() ? 'password' : 'text' }
+					type={ isValueHidden ? 'password' : 'text' }
 				/>
 
-				<span className={ toggleVisibilityClasses } onClick={ this.togglePasswordVisibility }>
-					{ this.hidden() ? <Icon icon={ unseen } /> : <Icon icon={ seen } /> }
-				</span>
+				<Button
+					className={ clsx( {
+						'form-password-input__toggle': true,
+						'form-password-input__toggle-visibility': ! hideToggle,
+					} ) }
+					onClick={ this.togglePasswordVisibility }
+					aria-hidden={ isHidden }
+					tabIndex={ isHidden ? -1 : undefined }
```

Comment:
Ensures the button is not accessible when this field is in the DOM but hidden visually